src/System.Private.CoreLib/shared/System/Runtime/ExceptionServices/ExceptionDispatchInfo.cs

```diff
@@ -102,7 +31,7 @@ public static ExceptionDispatchInfo Capture(Exception source)
         {
             if (source == null)
             {
-                throw new ArgumentNullException(nameof(source), SR.ArgumentNull_Obj);
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
```

Comment:
We use ThrowHelper for performance critical paths only. This one is not performance critical path. I have pushed the change to fix this already.